File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int elapsed)
    {
        return this.ExpectedMinutesInOven() - elapsed;
    }

    public int PreparationTimeInMinutes(int layers)
    {
        return layers * 2;
    }

    public int ElapsedTimeInMinutes(int layers, int elapsedTimeInOven)
    {
        return this.PreparationTimeInMinutes(layers) + elapsedTimeInOven;
    }
}

File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven elapsed = expectedMinutesInOven - elapsed

let preparationTimeInMinutes layers = layers * 2

let elapsedTimeInMinutes layers timeInOven = 
    let prepTime = preparationTimeInMinutes layers
    prepTime + timeInOven
